### PR TITLE
Don't remove references to `DamFile` from blocks when copying a document from one scope to another if DAM scoping is not enabled

### PR DESCRIPTION
**Pull request by thomasdax98**
COM-658

### CODE REVIEW

**Comment by thomasdax98** [comment]
Generally, I think we need to rework the whole copy logic. It's way to unflexible, especially towards entites defined in the project. You should be able to define how custom entites should be handled when copying.

File: packages/admin/cms-admin/src/pages/pageTree/useCopyPastePages/sendPages.tsx [hunk 1]
```diff
@@ -413,16 +413,17 @@ function createPageTreeNodeIdReplacements(nodes: PageClipboard[]): ReplaceDepend
 function unhandledDependenciesFromDocument(
     documentType: DocumentInterface,
     document: GQLDocument,
-    { existingReplacements }: { existingReplacements: ReplaceDependencyObject[] },
+    { existingReplacements, hasDamScope = false }: { existingReplacements: ReplaceDependencyObject[]; hasDamScope?: boolean },
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/pages/pageTree/useCopyPastePages/sendPages.tsx:416]: It seems to me that you're missing passing this option when calling the function.

File: packages/admin/cms-admin/src/pages/pageTree/useCopyPastePages/sendPages.tsx [hunk 2]
```diff
@@ -413,16 +413,17 @@ function createPageTreeNodeIdReplacements(nodes: PageClipboard[]): ReplaceDepend
...
-        .dependencies(document)
-        .filter(
-            (dependency) =>
-                !existingReplacements.some(
-                    (replacement) => replacement.originalId === dependency.id && replacement.type === dependency.targetGraphqlObjectType,
-                ),
+    const unhandledDependencies = documentType.dependencies(document).filter((dependency) => {
+        if (dependency.targetGraphqlObjectType === "DamFile" && !hasDamScope) {
+            return false;
+        }
```
[the remark this answers is not in the document]
Comment (johnnyomair) [reply packages/admin/cms-admin/src/pages/pageTree/useCopyPastePages/sendPages.tsx:421]: Maybe add a comment explaining this?

File: packages/admin/cms-admin/src/pages/pageTree/useCopyPastePages/sendPages.tsx (hunk 1 above)
Comment (thomasdax98) [reply packages/admin/cms-admin/src/pages/pageTree/useCopyPastePages/sendPages.tsx:417]: Damn, sorry 🤦‍♂️

File: packages/admin/cms-admin/src/pages/pageTree/useCopyPastePages/sendPages.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (thomasdax98) [reply packages/admin/cms-admin/src/pages/pageTree/useCopyPastePages/sendPages.tsx]: Done

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**